### PR TITLE
Mon 4007 escalation configuration

### PR DESCRIPTION
**Pull request by bouda1**
Could we expect escalations to work?

### CODE REVIEW

File: src/configuration/applier/hostescalation.cc [hunk 1]
```diff
@@ -193,22 +163,16 @@ void applier::hostescalation::remove_object(
...
-    for (std::list<std::shared_ptr<engine::escalation> >::iterator
-             itt{escalations.begin()},
-         next_itt{escalations.begin()},
-         end{escalations.end()};
-         itt != end;
-         itt = next_itt) {
+       it != end; ++it) {
+    std::list<escalation*>& escalations(hit->second->get_escalations());
+    for (std::list<engine::escalation*>::iterator itt{escalations.begin()},
+         next_itt{escalations.begin()}, end{escalations.end()};
```
Comment (SylvestreG) [inline src/configuration/applier/hostescalation.cc]: if you need the next iterator somewhere, you can use std::next

File: src/configuration/applier/hostescalation.cc [hunk 2]
```diff
@@ -247,31 +216,39 @@ void applier::hostescalation::resolve_object(
...
-  uint64_t host_id{get_host_id(*obj.hosts().begin())};
-  host_id_map::iterator it{engine::host::hosts_by_id.find(host_id)};
-  for (std::list<std::shared_ptr<engine::escalation> >::const_iterator
-           itt{it->second->get_escalations().begin()},
-       end{it->second->get_escalations().end()};
-       itt != end; ++itt) {
+  // Find host escalation
+  bool found{false};
+  std::string const& hostname{*obj.hosts().begin()};
+  auto p(engine::hostescalation::hostescalations.equal_range(hostname));
```
Comment (SylvestreG) [inline src/configuration/applier/hostescalation.cc]: you left an auto here ;)

File: src/configuration/applier/serviceescalation.cc [hunk 3]
```diff
@@ -196,44 +174,51 @@ void applier::serviceescalation::modify_object(
...
-    std::list<std::shared_ptr<escalation>>& escalations(sit->second->get_escalations());
-    for (std::list<std::shared_ptr<engine::escalation>>::iterator
-        itt{escalations.begin()},
-        next_itt{escalations.begin()},
-        end{escalations.end()};
-        itt != end; itt = next_itt) {
+    std::list<escalation*>& escalations(sit->second->get_escalations());
+    for (std::list<engine::escalation*>::iterator itt{escalations.begin()},
+         next_itt{escalations.begin()}, end{escalations.end()};
+         itt != end; itt = next_itt) {
```
Comment (SylvestreG) [inline src/configuration/applier/serviceescalation.cc]: if you need next it you can use std::next.

File: src/configuration/applier/serviceescalation.cc [hunk 4]
```diff
@@ -251,32 +236,48 @@ void applier::serviceescalation::remove_object(
...
-        (*itt)->get_notification_interval() == obj.notification_interval() &&
-        (*itt)->get_escalation_period() == obj.escalation_period() &&
-        (*itt)->get_escalate_on(notifier::down) == (obj.escalation_options() & configuration::hostescalation::down) &&
-        (*itt)->get_escalate_on(notifier::unreachable) == (obj.escalation_options() & configuration::hostescalation::unreachable) &&
-        (*itt)->get_escalate_on(notifier::recovery) == (obj.escalation_options() & configuration::hostescalation::recovery)) {
+  // Find service escalation
+  bool found{false};
+  std::string const& hostname{*obj.hosts().begin()};
+  std::string const& desc{obj.service_description().front()};
+  auto p(engine::serviceescalation::serviceescalations.equal_range(
```
Comment (SylvestreG) [inline src/configuration/applier/serviceescalation.cc]: you left an auto here ;)

File: src/configuration/applier/hostescalation.cc (hunk 1 above)
Comment (bouda1) [reply src/configuration/applier/hostescalation.cc]: You are right :-)

File: src/configuration/applier/hostescalation.cc (hunk 2 above)
Comment (bouda1) [reply src/configuration/applier/hostescalation.cc]: Yes, the type returned by equal_range is horrible :-), one line of 80 characters is not enough :-D

**Pull request merged**